docs/dev_setup.rst

```diff
@@ -11,12 +11,58 @@ This section describes how someone wanting to contribute to the libCellML projec
+The libCellML codebase is hosted on GitHub and therefore Git is used to track changes in the codebase.  Before you begin you will need to have a few pre-requisites satisfied:
 
-#. GitHub user account (for the rest of this document we will call our user *andre*)
-#. Git installed
+#. GitHub user account (for the rest of this document we will call our user *andre*).
+#. Git installed.
+#. CMake installed.
+#. Toolchain for building software installed (dependent on the operating system).
 
-This document doesn't cover the process of satisfying these prerequisites, it is left as an exercise for you (Google is your friend).  The remainder of this document assumes that these prerequisites have been met.  This document covers setup from the command line if you are using a GUI like 'GitHub for Windows' then you will need to adjust the commands for the GUI you are using.
+This document provides a brief overview on how to satisfy these pre-requisites but it doesn't go into great detail (Google is your friend) as there are many great resources for acquiring and installing the required software.  In this document we will detail the software that we currently use but please be aware that software is always being updated and the documentation may not reflect the latest available.
+
+Pre-requisite acquisition
+=========================
+
+In this section we cover the retrieval and installation of pre-requisites.
+
+Git
+---
+
+Creating a GitHub user account is straightforward and can be done `here <https://github.com/join>`_.  Installing a *git* client is particular to each operating system and some pointers are offered below:
+
+* macOS *git* is pre-installed and available from the command line.
+* Ubuntu (and other linux distributions) *git* can be installed using the package manager with the command :code:`sudo apt-get install git`.
+* Windows *git* is available from a variety of vendors we commonly use `git for windows <http://gitforwindows.org/>`_, but other popular *git* implementations are:
+
+  - `GitHub Desktop <https://desktop.github.com/>`_
+  - `GitKracken <https://www.gitkraken.com/>`_
+  - `Git SCM <https://git-scm.com/>`_
+
+  The choice of implementation is one of personal preference.
+
+CMake
+-----
+
+*CMake* is the cross-platform family of tools designed to build, test and package software.  *CMake* is used to control the software compilation process using simple platform and compiler independent configuration files, and generate native makefiles and workspaces that can be used in the compiler environment of your choice.
+
+Again installation of *CMake* is particular to each operating system.  For Ubuntu (and other linux distributions) *CMake* can be installed using the package manager with the command :code:`sudo apt-get install cmake`.  For macOS and Windows *CMake* provides installation binaries from `here <https://cmake.org/download/>`_ choose the binary appropriate for your operating system and follow the installation instructions.
```

Comment:
Again, installation of *CMake* is particular to each operating system.  For Ubuntu (and other linux distributions), *CMake* can be installed using the package manager with the command :code:`sudo apt-get install cmake`.  For Windows and macOS, *CMake* provides installation binaries from `here <https://cmake.org/download/>`_. Choose the binary appropriate for your operating system and follow the installation instructions.